NEOSPlus/Components/Wizards/LightSourceWizardPlus.cs

```diff
@@ -0,0 +1,262 @@
+// MIT License
+
+// Copyright (c) 2023 marsmaantje
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE. 
+
+using BaseX;
+using FrooxEngine.UIX;
+using System;
+using FrooxEngine;
+using FrooxEngine.Undo;
+
+namespace LightSourcesWizardPlus
+{
+    [Category("Add-ons/Wizards")]
+    public class LightSourcesWizardPlus :
+      Component,
+      IDeveloperInterface
+    {
+        public readonly SyncRef<Slot> Root;
+        public readonly Sync<bool> ProcessPointLights;
+        public readonly Sync<bool> ProcessSpotLights;
+        public readonly Sync<bool> ProcessDirectionalLights;
+        public readonly Sync<bool> ProcessDisabled;
+        public readonly Sync<ShadowType> TargetShadowType;
+        public readonly Sync<bool> FilterColors;
+        public readonly Sync<color> Color;
+        protected readonly SyncRef<TextField> _tag;
+        protected readonly SyncRef<FloatTextEditorParser> _intensityField;
+        protected readonly SyncRef<FloatTextEditorParser> _rangeField;
+        protected readonly SyncRef<FloatTextEditorParser> _spotAngleField;
+        protected readonly SyncRef<FloatTextEditorParser> _maxColorVariance;
```

Comment:
Is this class meant to be inherited from? These ought to be private

Reply:
Not meant to be inherited from, changed them to private.